keyboards/onepiece/qk21/qk21.c

```diff
@@ -0,0 +1,91 @@
+/* Copyright 2021 linlin
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "qk21.h"
+
+#ifdef RGB_MATRIX_ENABLE
+led_config_t g_led_config = { {
+  // Key Matrix to LED Index
+  {  17,  18,  19, 20},
+  {  13,  14,  15, 16},
+  {  10,  11,  12, NO_LED},
+  {  6,  7,  8,  9 },
+  {  3,  4, 5, NO_LED },
+  {  0, NO_LED, 1, 2 }
+},{
+  // LED Index to Physical Position
+  { 0,  0 }, { 37,  0 }, { 75,  0 }, { 112,  0 },
+  { 0,  6 }, { 37,  6 }, { 75,  6 }, { 112,  6 },
+  { 0,  13 }, { 37,  13 }, { 75,  13 },
+  { 0,  19 }, { 37,  19 }, { 75,  19 }, { 112,  16 },
+  { 0,  25}, { 37,  25}, { 75,  25 },
+  { 18, 32 }, { 75,  32 }, { 75, 29 }
+
+}, {
+  // LED Index to Flag
+  4, 4, 4, 4,
+  4, 4, 4, 4,
+  4, 4, 4, 4,
+  4, 4, 4, 4,
+  4, 4, 4, 4,
+  4
+}};

```

Comment:
This structure should be moved to the info.json file. 

You can run `qmk info -kb onepiece/qk21 -f json` to generate the config, automatically.